codec-redis/src/main/java/io/netty/handler/codec/redis/RedisDecoder.java

```diff
@@ -0,0 +1,232 @@
+        if (lineByteBuf == null) {
+            return false;
+        }
+        final long length = parseRedisNumber(lineByteBuf);
+        if (type == RedisMessageType.ARRAY_HEADER) {
+            addMessageToOut(out, RedisMessageFactory.createArrayHeader(length));
+            return true;
+        } else if (type == RedisMessageType.BULK_STRING) {
+            bulkStringLength = length;
+            state = State.DECODE_BULK_STRING;
+            return true;
+        } else {
+            throw new CodecException("bad type: " + type);
+        }
+    }
+
+    private boolean decodeBulkString(ByteBuf in, List<Object> out) throws Exception {
+        if (bulkStringLength == NULL_RESPONSE) {
+            // $-1\r\n
+            addMessageToOut(out, RedisMessageFactory.nullBulkString());
+        } else if (bulkStringLength == 0L) {
+            // $0\r\n <here> \r\n
+            if (in.readableBytes() < CRLF_LENGTH) {
+                return false;
+            }
+            in.skipBytes(CRLF_LENGTH);
+            addMessageToOut(out, RedisMessageFactory.emptyBulkString());
+        } else if (bulkStringLength > 0L) {
+            // ${bulkStringLength}\r\n <here> {data...}\r\n
+            if (in.readableBytes() < bulkStringLength + CRLF_LENGTH) {
+                return false;
+            }
+            ByteBuf content = in.readSlice((int) bulkStringLength);
+            in.skipBytes(CRLF_LENGTH);
+            addMessageToOut(out, RedisMessageFactory.createBulkString(content.retain()));
+        } else {
+            throw new CodecException("bad bulkStringLength: " + bulkStringLength);
+        }
+        return true;
+    }
+
+    private void addMessageToOut(List<Object> out, RedisMessage msg) {
+        out.add(msg);
+        resetDecoder();
+    }
+
+    private RedisMessage newInlineRedisMessage(RedisMessageType messageType, ByteBuf bytes) {
+        switch (messageType) {
+        case SIMPLE_STRING:
+            return RedisMessageFactory.createSimpleString(bytes);
+        case ERROR:
+            return RedisMessageFactory.createError(bytes);
+        case INTEGER:
+            return RedisMessageFactory.createInteger(parseRedisNumber(bytes));
+        default:
+            throw new CodecException("bad type: " + messageType);
+        }
+    }
+
+    private static ByteBuf readLine(ByteBuf in) {
+        final int lfIndex = in.forEachByte(ByteProcessor.FIND_LF);
+        if (lfIndex < 0) {
+            return null;
+        }
+        final int length = lfIndex - in.readerIndex() - 1; // `-1` is for CR
+        return readBytes(in, length);
+    }
+
+    private static ByteBuf readBytes(ByteBuf in, int length) {
+        ByteBuf data = in.readSlice(length);
+        in.skipBytes(CRLF_LENGTH); // skip CRLF
+        return data;
+    }
+
+    private long parseRedisNumber(ByteBuf byteBuf) {
+        toLongProcessor.result = 0;
+        toLongProcessor.first = true;
+        byteBuf.forEachByte(toLongProcessor);
```

Comment:
We should detect if we have "overflowed" the legitimate range of numbers. For example we should not parse more than 64 characters for a positive number and 65 characters for a negative number.  It looks like `byteBuf` is already a slice of what we expect to be the number ... so perhaps we can validate the length before we iterate?